framework/PostProcessors/InterfacedPostProcessor.py

```diff
@@ -157,69 +187,66 @@ def collectOutput(self, finishedJob, output):
-            if key in output.getParaKeys('inputs'):
-              output.updateInputValue(key,exportDict['inputSpaceParams'][key])
-          for key in exportDict['outputSpaceParams']:
-            if key in output.getParaKeys('outputs'):
-              output.updateOutputValue(key,exportDict['outputSpaceParams'][str(key)])
-      for key in exportDict['metadata']:
-        output.updateMetadata(key,exportDict['metadata'][key])
-    else:
-      # output.type == 'PointSet':
-      for key in exportDict['inputSpaceParams']:
-        if key in output.getParaKeys('inputs'):
-          for value in exportDict['inputSpaceParams'][key]:
-            output.updateInputValue(str(key),value)
-      for key in exportDict['outputSpaceParams']:
-        if str(key) in output.getParaKeys('outputs'):
-          for value in exportDict['outputSpaceParams'][key]:
-            output.updateOutputValue(str(key),value)
-      for key in exportDict['metadata']:
-        output.updateMetadata(key,exportDict['metadata'][key])
-
-
-  def inputToInternal_OLD(self,input):
+  def returnFormat(self,location):
+    if location == 'input':
+      return self.inputFormat
+    elif location=='output':
+      return self.outputFormat
+
+  def inputToInternal_OLD(self,inputs):
     """
       Function to convert the received input into a format this object can
       understand
       @ In, input, list, list of dataObjects handed to the post-processor
       @ Out, inputDict, list, list of dictionaries this object can process
     """
     inputDict = []
-    for inp in input:
+    for inp in inputs:
       if type(inp) == dict:
         return [inp]
       else:
-        inputDictTemp = {'data':{}, 'metadata':{}}
-        inputDictTemp['data']['input']  = copy.deepcopy(inp.getInpParametersValues())
-        inputDictTemp['data']['output'] = copy.deepcopy(inp.getOutParametersValues())
-        inputDictTemp['metadata']       = copy.deepcopy(inp.getAllMetadata())
-        inputDictTemp['name'] = inp.whoAreYou()['Name']
-        inputDictTemp['type'] = str(inp.type)
+        inputDictTemp = {}
+        inputDictTemp['inpVars']   = inp.getVars('input')
+        inputDictTemp['outVars']   = inp.getVars('output')
+        inputDictTemp['data']      = inp.asDataset(outType='dict')['data']
+        inputDictTemp['dims']      = inp.getDimensions('output')
+        inputDictTemp['type']      = inp.type
+        inputDictTemp['numberRealizations'] = len(inp)
+        self.metaKeys = inp.getVars('meta')
+        for key in self.metaKeys:
+          try:
+            inputDictTemp['data'][key]  = inp.getMeta(pointwise=True,general=True)[key].values
+          except:
+            self.raiseADebug('The following key: ' + str(key) + ' has not passed to the Interfaced PP')
+        inputDictTemp['name']     = inp.name
         inputDict.append(inputDictTemp)
     return inputDict
 
-  def inputToInternal(self,input):
+  def collectOutput(self, finishedJob, output):
```

Comment:
Am I reading this right, that there's two `collectOutput` and two `returnFormat` defined in this class?

Reply:
I removed the useless ones